### PR TITLE
Fix torrent.done state when there are deselected files (fix #316)

### LINKED ISSUE #316: Event 'done' isn't called when there are deselected files
This checks for all the files to be downloaded instead it should check only for the "selected" files/pieces.
https://github.com/feross/webtorrent/blob/master/lib/storage.js#L597

So when all the wanted files are downloaded the torrent wont be done if there are deselected files.

### PR DESCRIPTION
**Pull request by feross**
Supersedes PR https://github.com/feross/webtorrent/pull/533

### CODE REVIEW

File: lib/torrent.js [hunk 1]
```diff
@@ -1154,8 +1154,20 @@ Torrent.prototype._checkDone = function () {
...
+    var selection = self._selections[i]
+    for (var piece = selection.from; piece <= selection.to; piece++) {
+      if (!self.bitfield.get(piece)) {
+        done = false
+        break
+      }
+    }
+    if (!done) break
+  }
+  if (!self.done && done) {
```
Comment (feross) [inline lib/torrent.js]: This is an extra safety-net to ensure we don't emit 'done' twice.

File: lib/torrent.js [hunk 2]
```diff
@@ -1154,8 +1154,20 @@ Torrent.prototype._checkDone = function () {
...
-  if (self.files.every(function (file) { return file.done })) {
+  // is the torrent done? (if all current selections are satisfied, or there are
+  // no selections, then torrent is done)
+  var done = true
+  for (var i = 0; i < self._selections.length; i++) {
+    var selection = self._selections[i]
+    for (var piece = selection.from; piece <= selection.to; piece++) {
+      if (!self.bitfield.get(piece)) {
+        done = false
+        break
```
Comment (feross) [inline lib/torrent.js]: This isn't strictly necessary – just an optimization

File: lib/torrent.js [hunk 3]
```diff
@@ -1154,8 +1154,20 @@ Torrent.prototype._checkDone = function () {
...
+      }
+    }
+    if (!done) break
```
Comment (feross) [inline lib/torrent.js]: Same here – this is just an optimization

**Comment by DiegoRBaquero** [comment]
Looks good

**Comment by feross** [comment]
Released as 0.65.1.

**Pull request merged**